src/finalization/state_repository.cpp

```diff
@@ -129,6 +129,7 @@ void RepositoryImpl::TrimUntilHeight(blockchain::Height height) {
       assert(index != nullptr);
     }
     if (static_cast<blockchain::Height>(index->nHeight) < height) {
+      m_state_db->Erase(*it->first);
```

Comment:
Can this under any circumstances fail/throw an exception?

Reply:
`m_state_db->Erase`  can return false in case when there was no such state on the disk. But it's a normal behavior as data in-memory and on-disk are not necessarily synchronous (we flush data on disk occasionally from `validation.cpp/FlushStateToDisk`).